modin/pandas/test/test_general.py

```diff
@@ -105,50 +114,66 @@ def test_merge():
-        # Defaults
-        modin_result = pd.merge(modin_df, modin_df2, how=how)
+    join_type_outer = "outer"
+    join_type_inner = "inner"
+    for how in (join_type_outer, join_type_inner):
+        if how == join_type_outer:
+            warning_catcher = warns_that_defaulting_to_pandas()
+        else:
+            warning_catcher = contextlib.nullcontext()
+        with warning_catcher:
+            modin_result = pd.merge(modin_df, modin_df2, how=how)
         pandas_result = pandas.merge(pandas_df, pandas_df2, how=how)
         df_equals(modin_result, pandas_result)
 
         # left_on and right_index
-        modin_result = pd.merge(
-            modin_df, modin_df2, how=how, left_on="col1", right_index=True
-        )
+        with warns_that_defaulting_to_pandas():
+            modin_result = pd.merge(
+                modin_df, modin_df2, how=how, left_on="col1", right_index=True
+            )
         pandas_result = pandas.merge(
             pandas_df, pandas_df2, how=how, left_on="col1", right_index=True
         )
         df_equals(modin_result, pandas_result)
 
         # left_index and right_on
-        modin_result = pd.merge(
-            modin_df, modin_df2, how=how, left_index=True, right_on="col1"
-        )
+        with warns_that_defaulting_to_pandas():
+            modin_result = pd.merge(
+                modin_df, modin_df2, how=how, left_index=True, right_on="col1"
+            )
         pandas_result = pandas.merge(
             pandas_df, pandas_df2, how=how, left_index=True, right_on="col1"
         )
         df_equals(modin_result, pandas_result)
 
         # left_on and right_on col1
-        modin_result = pd.merge(
-            modin_df, modin_df2, how=how, left_on="col1", right_on="col1"
-        )
+        if how == join_type_outer:
+            warning_catcher = warns_that_defaulting_to_pandas()
+        else:
+            warning_catcher = contextlib.nullcontext()
+        with warning_catcher:
+            modin_result = pd.merge(
+                modin_df, modin_df2, how=how, left_on="col1", right_on="col1"
+            )
         pandas_result = pandas.merge(
             pandas_df, pandas_df2, how=how, left_on="col1", right_on="col1"
         )
         df_equals(modin_result, pandas_result)
 
         # left_on and right_on col2
-        modin_result = pd.merge(
-            modin_df, modin_df2, how=how, left_on="col2", right_on="col2"
-        )
+        if how == join_type_outer:
+            warning_catcher = warns_that_defaulting_to_pandas()
+        else:
+            warning_catcher = contextlib.nullcontext()
+        with warning_catcher:
+            modin_result = pd.merge(
+                modin_df, modin_df2, how=how, left_on="col2", right_on="col2"
+            )
         pandas_result = pandas.merge(
             pandas_df, pandas_df2, how=how, left_on="col2", right_on="col2"
         )
         df_equals(modin_result, pandas_result)
 
-        # left_index and right_index
```

Comment:
why removed? Instead of removing, I would propose to make the comment more verbose, if you feel the need to do that

Reply:
I restored it.